apps/arweave/src/ar_nonce_limiter.erl

```diff
@@ -1090,17 +1093,30 @@ apply_external_update2(Update, State) ->
 %% Note: an important job of this function is to ensure that VDF steps are only processed once.
 %% We truncate Session.steps such the previously processed steps are not sent to
 %% send_events_for_external_update.
-apply_external_update3(State, SessionKey, CurrentSessionKey, Session, Steps) ->
-	#state{ last_external_update = {Peer, _} } = State,
-	?LOG_DEBUG([{event, apply_external_vdf},
-		{result, ok},
+apply_external_update3(State, SessionKey, Session, Steps) ->
+	#state{ last_external_update = {Peer, _}, current_session_key = CurrentSessionKey } = State,
+
+	?LOG_DEBUG([{event, new_vdf_step}, {source, apply_external_vdf},
 		{vdf_server, ar_util:format_peer(Peer)},
 		{session_key, encode_session_key(SessionKey)},
 		{step_number, Session#vdf_session.step_number},
 		{length, length(Steps)}]),
-	State2 = cache_session(State, SessionKey, CurrentSessionKey, Session),
+
+	{_, SessionInterval, _} = SessionKey,
+	CurrentSessionInterval = case CurrentSessionKey of
+		{_, Interval, _} -> Interval;
+		_ -> -1
+	end,
+	State2 = case SessionInterval > CurrentSessionInterval of
```

Comment:
The current design is such that assigning the current session has to be consensus-based only (i.e. happen only in apply_tip). What is the motivation for this change?

Reply:
Ah, got it. The issue this was try to solve was the following:
1. Sometimes the VDF server would update its current session before the VDF client
2. The client would receive those updates, but wouldn't mine against them since the session was neither the current, nor previous session (i.e. it treated the updates as stale updates)
3. This caused the miner to have periods of 0 hashrate

I can find another way to address the issue without breaking consensus!

Reply:
Fixed!